### PR TITLE
Updated README.md and Fixed Crash that occurs when period is set too high.

### PR DESCRIPTION
**Pull request by barrymichaeldoyle**
Hey Rick :D

Sorry about all the line changes, my vscode settings might have done some auto formatting. Are you using vscode? If so, you should probably create a project level config for it 👍 

Anyway, I've updated the README.md file to include running npm install (for users who generally forget to install node modules :)

I also fixed the bug where setting period too high crashes the app.

Basically when period is set too high, a lot of it's cases are undefined. So I wrote some fallback code for all those cases. Another solution might be to format period to match the highest possible value or lowest possible value (if they went too low) upon periodChange.

Hope this is useful :D

This project is really cool :)

### CODE REVIEW

**Comment by rickkln** [comment]
Hey Barry,

Thanks for this! Those fixes are super useful!

Before I merge I just wanna get the code style issues sorted. I am using VS Code, but trying to avoid checking in IDE related files into the project (just to keep it IDE agnostic). I was assuming that the ESLint and TS configs would be enough to get things consistent. If I checkout your branch I see a bunch of ESLint errors, so maybe VS Code isn't picking up the project's `.eslintrc.js`? You should be able to fix those in VS Code by doing **⌘⇧P** and then running `ESLint: Fix all auto-fixable Problems`.

Note, there are a couple ESLint errors in the file you were working in that are fine to be there (1. it complains about the styles import, and 2. it complains about types for some of the tables, on this one in the end I'm not happy with the ts support in react-table and will probably just avoid it in the future).

Let me know if you can see what is happening with ESLint your side? I'll include a screenshot of an example of the issues on your branch, and then the issues after fixing the auto fixable ones.

Cheers,
Rick

Screenshot 2020-05-08 at 10 06 15
Screenshot 2020-05-08 at 10 20 28

**Comment by barrymichaeldoyle** [comment]
Cool man, I'll go sort this stuff out later this afternoon and update my PR :)

**Comment by rickkln** [comment]
Thanks man, no rush!

**Comment by barrymichaeldoyle** [comment]
Cool, I decided to just go through and sort it out :)

The fix isn't ideal, I just set the React.memo columns to be imported as the expected type to satisfy the react-table (it works). I did the same thing for the initialStates.

Then I also added an index.d.ts file to resolve those pesky errors around the module.css imports :D

File: src/components/shared/tables/tables.tsx [hunk 1]
```diff
@@ -9,47 +9,56 @@ import { getCSSClassFor, getPeriodName } from '../../../utilities/periodUtils';
...
 };
 
 const stickyGlobal = (row: Row, desc: boolean, value: number) => {
-  const global = desc
-    ? 1
-    : -1;
-  return row.values.name === 'Global'
-    ? global
-    : value;
+  const global = desc ? 1 : -1;
```
Comment (rickkln) [inline src/components/shared/tables/tables.tsx:26]: Ternary style in the project is to always enforce multiline ternaries with the format like so:
```
condition
  ? resultA
  : resultB;
```
Seems silly for simple cases like this, but the idea is basically to just avoid 3 expressions in one line, and make the rule simple for consistency.

File: src/components/shared/tables/tables.tsx [hunk 2]
```diff
@@ -449,21 +547,14 @@ export const GrowthSummaryTable = ({ data }: { data: Country[] }) => {
...
 
   return (
-    <div
-      className={
-        data.length === 1
-          ? styles.tinyTable
-          : ''
-      }
-    >
+    <div className={data.length === 1 ? styles.tinyTable : ''}>
```
Comment (rickkln) [inline src/components/shared/tables/tables.tsx:557]: Same multiline ternary note.

**Review by rickkln — APPROVED** [review]
Amazing! Thanks for the fixes to the ts errors, super useful! Left some comments, though they're not serious so I approved so long.

Looking at the actual fix for the crash it's quite annoying that those values are coming out as undefined at that point. Would have thought they would be checked before being passed in to those callbacks, again if only react-table had better types would have been able to catch that at build time 😬 Probably should fix it in the data utils but will do that another time 😅

**Comment by barrymichaeldoyle** [comment]
Cool man, glad I could help :)

Yeah I didn’t really dig too much into the source of the problem, so my
work pretty much just addressed the symptoms.

I’ll make another PR tomorrow on changing those fixes to ternary styles :)

**Comment by rickkln** [comment]
Gonna merge this so long, I'll tweak the ternaries don't worry :)

**Comment by barrymichaeldoyle** [comment]
Epic, thanks man :)

**Pull request merged**